plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -33,6 +34,68 @@
 
 logger = sgtk.LogManager.get_logger(__name__)
 
+class ProgressHandler(QtCore.QObject):
+    """
+    An object that wraps a QTimer that is used to periodically check
+    for updates that need to be made to the progress bar in Maya. This
+    will always execute progress updates on the main thread.
+    """
+    PROGRESS_INTERVAL = 150 # milliseconds
+
+    def __init__(self):
+        try:
+            import shiboken2 as shiboken
+        except ImportError:
+            import shiboken
+
+        ptr = OpenMayaUI.MQtUtil.mainWindow()
+        parent = shiboken.wrapInstance(long(ptr), QtGui.QMainWindow)
+
+        super(ProgressHandler, self).__init__(parent=parent)
+
+        self._progress_value = None
+        self._message = None
+        self._timer = QtCore.QTimer(parent=self)
+
+        self._timer.timeout.connect(self._set_progress)
+        self._timer.start(self.PROGRESS_INTERVAL)
+
+    @property
+    def timer(self):
+        """
+        The QTimer instance that's updating progress.
+        """
+        return self._timer
+
+    def _set_progress(self):
+        """
+        Sets progress. Must be run from the main thread!
+        """
+        if self._message and self._progress_value:
+            _show_progress_bar(self._progress_value, self._message)
+            self._message = None
+            self._progress_value = None
+
+    def _handle_bootstrap_progress(self, progress_value, message):
+        """
+        Callback function that reports back on the toolkit and engine bootstrap progress.
+
+        This function is executed in the main thread by the main event loop.
+
+        :param progress_value: Current progress value, ranging from 0.0 to 1.0.
+        :param message: Progress message to report.
+        """
+
+        logger.debug("Bootstrapping Shotgun: %s" % message)
+
+        # Show the progress bar, and update its value and message.
+        # _show_progress_bar(progress_value, message)
+        self._progress_value = progress_value
+        self._message = message
+
+
+PROGRESS_HANDLER = ProgressHandler()
```

Comment:
Could you lower case this? Only constants should be in upper case.

Reply:
👍